sycl/include/CL/sycl/handler.hpp

```diff
@@ -128,6 +129,39 @@ struct check_fn_signature<F, RetT(Args...)> {
 
 __SYCL_EXPORT device getDeviceFromHandler(handler &);
 
+#if defined(__SYCL_ID_QUERIES_FIT_IN_INT__)
+template <typename T> struct NotIntMsg;
+
+// TODO reword for "`fsycl-id-queries-fit-in-int' optimization flag." when
+// implemented
+template <int Dims> struct NotIntMsg<range<Dims>> {
+  constexpr static char *Msg = "Provided range is out of integer limits. "
+                               "Pass `-U__SYCL_ID_QUERIES_FIT_IN_INT__' to "
+                               "disable range check.";
+};
+
+template <int Dims> struct NotIntMsg<id<Dims>> {
+  constexpr static char *Msg = "Provided offset is out of integer limits. "
+                               "Pass `-U__SYCL_ID_QUERIES_FIT_IN_INT__' to "
+                               "disable offset check.";
+};
+#endif
+
+template <int Dims, typename T>
+typename std::enable_if<std::is_same<T, range<Dims>>::value ||
+                        std::is_same<T, id<Dims>>::value>::type
+checkValueRange(const T &V) {
+#if defined(__SYCL_ID_QUERIES_FIT_IN_INT__)
+  static constexpr size_t Limit =
+      static_cast<size_t>(std::numeric_limits<int>::max());
```

Comment:
```suggestion
      static_cast<size_t>((std::numeric_limits<int>::max)());
```

Reply:
There is a known issue with max macro defined in windows.h
and the suggested fix is the easiest fix for that.

Reply:
I am vomiting... :-)

Reply:
> I am vomiting... :-)

Is it something special with the patch or just bad luck?

Reply:
It was just reacting to the need to add a `)` after `max` to prevent a macro to trigger on Windows... :vomiting_face:

This kind of code would deserve at least a comment :smile: